Backend/NoCode.FlowerShop.Application/FlowerArrangements/Create/CreateFlowerArrangementHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using ErrorOr;
+using MediatR;
+using NoCode.FlowerShop.Application.Common.Interfaces.Persistence;
+using NoCode.FlowerShop.Domain;
+using NoCode.FlowerShop.Domain.Common.ErrorsCollection;
+
+namespace NoCode.FlowerShop.Application.FlowerArrangements.Create;
+internal class CreateFlowerArrangementHandler :
+    IRequestHandler<CreateFlowerArrangementCommand, ErrorOr<CreateFlowerArrangementResult>>
+{
+    private readonly IFlowerArrangementRepository _flowerArrangementRepository;
+
+    private readonly IFlowerArrangementCategoryRepository _flowerArrangementCategoryRepository;
+
+    private readonly IFlowerRepository _flowerRepository;
+
+    public CreateFlowerArrangementHandler(
+        IFlowerArrangementRepository flowerArrangementRepository, 
+        IFlowerArrangementCategoryRepository flowerArrangementCategoryRepository, 
+        IFlowerRepository flowerRepository)
+    {
+        _flowerArrangementRepository=flowerArrangementRepository;
+        _flowerArrangementCategoryRepository=flowerArrangementCategoryRepository;
+        _flowerRepository=flowerRepository;
+    }
+
+    public async Task<ErrorOr<CreateFlowerArrangementResult>> Handle(CreateFlowerArrangementCommand request, CancellationToken cancellationToken)
+    {
+        var flowerArrangement = await _flowerArrangementRepository.GetByNameAsync(request.Name, cancellationToken);
+        if (flowerArrangement is not null)
+            return Errors.FlowerArrangement.DuplicateName;
+
+        var category = await _flowerArrangementCategoryRepository.GetById(request.CategoryId, cancellationToken);
+        if (category is null)
+            return Errors.FlowerArrangement.CategoryNotFound;
+
+        var flowersIds = request.Flowers.Select(x => x.Id).ToList();
+        var flowers = _flowerRepository
+            .GetAll()
+            .Where(x => flowersIds.Contains(x.Id))
+            .ToList();
+
+        var hasExistingFlowersIds = flowers.Count == flowersIds.Count;
+
+        if (flowers == null || !hasExistingFlowersIds)
+            return Errors.FlowerArrangement.FlowerNotFound;
+
+        var flowerArrangementFlowers = new List<Domain.Flowers>();
+        foreach(var flower in flowers)
+        {
+            var quantity = request.Flowers
+                .Where(x => x.Id == flower.Id)
+                .First()
+                .Quantity;
+
+            flowerArrangementFlowers.Add(new(flower, quantity));
+        }
```

Comment:
If you need such complex foreach loop, perhaps those `flowers` and `flowersid` should have been not lists, but maybe dictionaries? Or other more convenient data structure?

Reply:
So I've checked whether it is possible to pass a dictionary type from JSON body, and yes it is, but I didn't like how the structure of JSON body will look like then.

Otherwise it is needed to somehow convert the passed `FlowerDto` (or whatever name we agree on) to a dictionary type and I believe it will only save one line of code (+ some insignificant amount of efficiency). So maybe let's leave it as it is for now?

If we want to use dictionary, then my proposal is to convert `FlowerDto` to `Dictionary<string, int>` when converted into `CreateFlowerArrangementCommand`.